salt/utils/zeromq.py

```diff
@@ -3,21 +3,60 @@
 # Import Python libs
 from __future__ import absolute_import
 
-# Import Salt libs
+import logging
+import tornado.ioloop
 from salt.exceptions import SaltSystemExit
 
-# Import 3rd-party libs
+log = logging.getLogger(__name__)
+
 try:
     import zmq
-    HAS_ZMQ = True
 except ImportError:
-    HAS_ZMQ = False
+    zmq = None
+    log.debug('ZMQ module is not found')
+
+ZMQDefaultLoop = None
+ZMQ_VERSION_INFO = (-1, -1, -1)
+LIBZMQ_VERSION_INFO = (-1, -1, -1)
+
+try:
+    if zmq:
+        ZMQ_VERSION_INFO = tuple([int(v_el) for v_el in zmq.__version__.split('.')])
+        LIBZMQ_VERSION_INFO = tuple([int(v_el) for v_el in zmq.zmq_version().split('.')])
+        if ZMQ_VERSION_INFO[0] > 16:  # 17.0.x+ deprecates zmq's ioloops
+            ZMQDefaultLoop = tornado.ioloop.IOLoop
+except Exception:
+    log.exception('Error while getting LibZMQ/PyZMQ library version')
+
+if ZMQDefaultLoop is None:
+    try:
+        import zmq.eventloop.ioloop
+        # Support for ZeroMQ 13.x
+        if not hasattr(zmq.eventloop.ioloop, 'ZMQIOLoop'):
+            zmq.eventloop.ioloop.ZMQIOLoop = zmq.eventloop.ioloop.IOLoop
+        if tornado.version_info < (5,):
+            ZMQDefaultLoop = zmq.eventloop.ioloop.ZMQIOLoop
+    except ImportError:
+        ZMQDefaultLoop = None
+    if ZMQDefaultLoop is None:
+        ZMQDefaultLoop = tornado.ioloop.IOLoop
+
+
+def install_zmq():
+    '''
+    While pyzmq 17 no longer needs any special integration for tornado,
+    older version still need one.
+    :return:
+    '''
+    if zmq and ZMQ_VERSION_INFO[0] < 17:
+        if tornado.version_info < (5,):
```

Comment:
Style suggestion: Combine these two "if" statements into one. eg:
```
if zmq and ZMQ_VERSION_INFO[0] < 17 and tornado.version_info < (5,):
    zmq.eventloop.ioloop.install()
```

Reply:
I was actually backporting the other PR. This went alongside with your fixes. Additionally, I added another `if`, because there are other issues fixes are coming soon, something like "if not zmq-17, but tornado-5" etc, hence your suggestion will opt out all that.